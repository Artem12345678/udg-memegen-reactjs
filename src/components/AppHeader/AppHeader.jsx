import React from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import { Link } from "react-router-dom";

import "./AppHeader.scss";

const AppHeader = () => {
  return (
    <header className="header bg-primary">
      <div className="container">
        <Navbar variant="dark" expand="lg">
          <Navbar.Brand as={Link} to="/" className="brand">
            MemeGen
          </Navbar.Brand>

          <Navbar.Toggle aria-controls="nav-collapse" />

          <Navbar.Collapse id="nav-collapse">
            <Nav className="ml-auto">
              <Nav.Link as={Link} to="/">
                Home
              </Nav.Link>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      </div>
    </header>
  );
};

export default AppHeader;
