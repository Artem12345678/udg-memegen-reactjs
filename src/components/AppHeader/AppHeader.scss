.header {
  padding: 10px 0;
}

.brand {
  font-size: 56px !important;
  font-weight: 700 !important;
  display: block;
  letter-spacing: -1.4px !important;
  color: #fff !important;
}
