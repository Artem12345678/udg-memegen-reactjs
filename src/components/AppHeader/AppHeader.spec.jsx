import React from "react";
import { shallow } from "enzyme";

import AppHeader from "./AppHeader";

describe("AppHeader.jsx", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AppHeader />);
  });

  it("matches snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
