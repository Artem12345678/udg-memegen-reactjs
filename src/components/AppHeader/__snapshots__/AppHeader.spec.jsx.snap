// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader.jsx matches snapshot 1`] = `
<header
  className="header bg-primary"
>
  <div
    className="container"
  >
    <Navbar
      collapseOnSelect={false}
      expand="lg"
      variant="dark"
    >
      <NavbarBrand
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        className="brand"
        to="/"
      >
        MemeGen
      </NavbarBrand>
      <NavbarToggle
        aria-controls="nav-collapse"
        label="Toggle navigation"
      />
      <NavbarCollapse
        id="nav-collapse"
      >
        <Nav
          className="ml-auto"
          fill={false}
          justify={false}
        >
          <NavLink
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "Link",
                "propTypes": Object {
                  "innerRef": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              }
            }
            disabled={false}
            to="/"
          >
            Home
          </NavLink>
        </Nav>
      </NavbarCollapse>
    </Navbar>
  </div>
</header>
`;
